### ISSUE TITLE
Uncaught ReferenceError: fullpage is not defined

### ISSUE THREAD

**Issue by xiaotiandada** [alvarotrigo/vue-fullpage.js#60]
package.json
    "vue-fullpage.js": "^0.0.6"

I cloned the demo(demo/webpack), but this problem still occurs....

**Comment by funkyfuture** [comment]
don't know if this is exactly the same issue, but here are more details of my experiences:

1. download and extract the demo app
2. `npm i`
3. `npm run dev`
4. open the app in browser

```
Uncaught ReferenceError: fullpage is not defined
    at Object.eval (webpack-internal:///36:290)
    at __webpack_require__ (webpack-internal:///36:42)
    at disposed (webpack-internal:///36:88)
    at eval (webpack-internal:///36:91)
    at webpackUniversalModuleDefinition (webpack-internal:///36:15)
    at eval (webpack-internal:///36:22)
    at Object.<anonymous> (build.js:965)
    at __webpack_require__ (build.js:679)
    at fn (build.js:89)
    at eval (webpack-internal:///32:5)
    at Object.<anonymous> (build.js:941)
    at __webpack_require__ (build.js:679)
    at fn (build.js:89)
    at Object.<anonymous> (build.js:790)
    at __webpack_require__ (build.js:679)
    at build.js:725
    at build.js:728
```

OS is a Ubuntu Bionic w/ Chromium 69.0.3497.81 & Firefox 62.0.3

**Comment by alvarotrigo** [comment]
Please provide an isolated reproduction in [jsfiddle](https://jsfiddle.net/alvarotrigo/ea17skjr/) or [Codepen](https://codepen.io/alvarotrigo/pen/qqabrp), with no CSS or JS files external to fullPage.js and the minimum amount of HTML code. Use empty sections unless strictly necessary for the reproduction.

Perhaps you might find useful to read my article regarding [how to create isolated reproductions](https://alvarotrigo.com/blog/how-not-suck-at-asking-coding-questions/).

**Comment by alvarotrigo** [comment]
The real problem is actually when running `npm run build` from `demos/webpack/`.
It throws this error:

> ERROR in build.js from UglifyJs
Unexpected token: operator (>) [./src/main.js:16,0][build.js:254,13]

But I'm not yet quite sure what's the cause of it.

**Comment by alvarotrigo** [comment]
There's an error when running `npm run build` from the root folder too. 
It doesn't seem to be detecting the babel transpiler? Any suggestions?

> ERROR in ./~/babel-loader/lib!./~/vue-loader/lib/selector.js?type=script&index=0!./src/FullPage.vue
Module build failed: SyntaxError: Unexpected token (67:10)

**Comment by alvarotrigo** [comment]
Your error is just a result of an error when using `npm run dev`. 
The root of the error is on the build process.

**Comment by funkyfuture** [comment]
this issue (https://github.com/webpack/webpack/issues/5548) might be related. but to my understanding it shouldn't as my environment has `acorn@5.7.3` installed.

**Comment by funkyfuture** [comment]
when i update to `webpack@^4.2.0`, i get the following on `npm run dev`

```
> demo@1.0.0 dev /tmp/webpack
> cross-env NODE_ENV=development webpack-dev-server --open --hot

module.js:550
    throw err;
    ^

Error: Cannot find module 'webpack/bin/config-yargs'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/tmp/webpack/node_modules/webpack-dev-server/bin/webpack-dev-server.js:54:1)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! demo@1.0.0 dev: `cross-env NODE_ENV=development webpack-dev-server --open --hot`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the demo@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by xiaoJinGong** [comment]
When i update to vue-fullpage.js@^0.0.7.
I get the following on npm run dev.

Uncaught ReferenceError: fullpage is not  defined

    at Object.eval (vue-fullpage.js?a45e:290)
    at __webpack_require__ (vue-fullpage.js?a45e:42)
    at disposed (vue-fullpage.js?a45e:88)
    at eval (vue-fullpage.js?a45e:91)
    at webpackUniversalModuleDefinition (vue-fullpage.js?a45e:15)
    at eval (vue-fullpage.js?a45e:22)
    at Object../node_modules/vue-fullpage.js/dist/vue-fullpage.js (app.js:3114)
    at __webpack_require__ (app.js:768)
    at fn (app.js:131)
    at eval (main.js:10)

**Comment by maxrisk** [comment]
I have the same issue...

**Comment by maxrisk** [comment]
@alvarotrigo You miss `import fullpage from 'fullpage.js'` at 'vue-fullpage.js/src/index.js'. 
@xiaotiandada You can fix this issue like I say.

**Comment by alvarotrigo** [comment]
> @alvarotrigo You miss `import fullpage from 'fullpage.js'` in 'vue-fullpage.js/src/index.js'.

@maxrisk nop. Fullpage is already included in `Fullpage.vue`:
See this line:

       import fullpage from 'fullpage.js/dist/fullpage.extensions.min';

Problem is clearly on the building process. It is not recognising ES6 syntax with babel for some reason.

**Comment by admin00001** [comment]
yes, i run it error as you,         i use the   @vue/cli 3.05     to crete the project . and i install th fullpage.js 0.07,then i import it within the main.js  like this:
```
import Vue from 'vue'
import App from './App.vue'

import VueFullPage from 'vue-fullpage.js'
Vue.use(VueFullPage)
Vue.config.productionTip = false

new Vue({
  render: h => h(App)
}).$mount('#app')

```

then i run yarn serve .it has error as 
```
vue-fullpage.js?a45e:290 Uncaught ReferenceError: fullpage is not defined
    at Object.eval (vue-fullpage.js?a45e:290)
    at __webpack_require__ (vue-fullpage.js?a45e:42)
    at disposed (vue-fullpage.js?a45e:88)
    at eval (vue-fullpage.js?a45e:91)
    at webpackUniversalModuleDefinition (vue-fullpage.js?a45e:15)
    at eval (vue-fullpage.js?a45e:22)
    at Object../node_modules/vue-fullpage.js/dist/vue-fullpage.js (app.js:1756)
    at __webpack_require__ (app.js:725)
    at fn (app.js:102)
    at eval (main.js:12)
```
then i search the code in   vue-fullpage.js?a45e:290 ,   the code is 
```
if (!window.fullpage_api) {
  window.fullpage = fullpage;
}
```
then  i  write the code
```
 console.log(window.fullpage)
 // undefinded 
// so the fullpage is not mounted on the global obeject window
```
may be you can use the @vue/cli to crete a project to find the reson.

**Comment by flex79** [comment]
I have the same issue after upgrading from 0.0.3 to 0.06 or 0.07. It does not seem to bundle fullpage.js somehow. 0.0.3 works fine. Using standard vue-cli setup

**Comment by xiaotiandada** [comment]
> I have the same issue after upgrading from 0.0.3 to 0.06 or 0.07. It does not seem to bundle fullpage.js somehow. 0.0.3 works fine. Using standard vue-cli setup

sad

**Comment by xiaotiandada** [comment]
sad

1. clone master/dev
2. open demos/webpack
3. npm i
4. npm run dev

---

result

```bash

Module build failed: ReferenceError: Unknown plugin "transform-vue-jsx" specified in "E:\\git\\clone\\vue-fullpage.js\\.babelrc" at 0, attempted to resolve relative to 

```

**Comment by Chiaparini** [comment]
@xiaotiandada 
you need to change the main.js reference of the example:
`import VueFullPage from '../node_modules/vue-fullpage.js/dist/vue-fullpage.js'`
the example seems to be referencing the root folder.

**Comment by Chiaparini** [comment]
ok I got it, I thought the demo folder was running with deployed version, sorry. But by running `npm i` on the root folder it seems to work

**Comment by alvarotrigo** [comment]
@Chiaparini yeap. But far from ideal isn't it?
Can't we just make it somehow so we won't need to do npm i on the root folder in addition to on the demo folder?

**Comment by Chiaparini** [comment]
well, for a quickfix I found out that when disabling the `.babelrc` file from babel loader of the webpack.config.js seems to work.

```
{
        test: /\.js$/,
        loader: 'babel-loader',
        options: {
          babelrc: false
        }
      },
```

**Comment by Chiaparini** [comment]
idk why the demo server is getting the root reference for the .babelrc though

**Comment by alvarotrigo** [comment]
I've opened a [new issue](https://github.com/alvarotrigo/vue-fullpage.js/issues/69) to follow up the new error reported by @xiaoJinGong https://github.com/alvarotrigo/vue-fullpage.js/issues/69

Please keep the new conversation in there.

**Comment by xiaotiandada** [comment]
> @xiaotiandada
> 你需要更改示例的main.js引用：
> `import VueFullPage from '../node_modules/vue-fullpage.js/dist/vue-fullpage.js'`
> 该示例似乎是引用根文件夹。

I try it    ~~~

**Comment by ankitsinghaniyaz** [comment]
@alvarotrigo I think we can make the `npm install` on the root not-required for the users. Here is how we can do that.

Inside the demo:

1. Add another webpack command `npm run dev-local` which will be used by maintainers to use the local copy of the code as you said in the comment above.
2. The normal `npm run dev` will use the remote latest version and should just required one `npm instal` by the users.

What do you think?

**Comment by alvarotrigo** [comment]
The issue has ben fixed on the dev branch thanks to this pull request https://github.com/alvarotrigo/vue-fullpage.js/pull/72 from @Chiaparini !

Thanks all for your suggestions!

**Comment by alvarotrigo** [comment]
Fixed in version 0.0.9

**Comment by xiaotiandada** [comment]
> Fixed in version 0.0.9

thanks meme
